aws/modules/register_group/dns.tf

```diff
@@ -0,0 +1,11 @@
+resource "aws_route53_record" "multi" {
+  count = "${length(var.registers)}"
```

Comment:
Actually this is still needed so we don't cause any downtime when we deploy these changes. This can be removed later as part of cleanup.

This PR creates a temporary hosted zone which contains the records deployed in the new way. I'll swap over the `NS` records on `openregister.org` manually to this new Hosted Zone before doing some cleanup, applying these DNS changes to the Hosted Zone that's created by the `core` Terraform module, and swapping the `NS` records back.